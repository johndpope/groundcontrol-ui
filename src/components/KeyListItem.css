.KeyListItem {
  margin: 1em 0;
}

.KeyListItem .header {
  padding-top: 0.5em;
}

.KeyListItem .description {
  margin: 1em 0;
}

.KeyListItem .description code {
  font: var(--mono-font);
  word-break: break-all;
  white-space: pre-wrap;
}
