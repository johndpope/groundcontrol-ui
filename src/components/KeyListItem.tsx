// Copyright 2019 Stratumn
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

import graphql from "babel-plugin-relay/macro";
import React from "react";
import {
  Button,
  List,
} from "semantic-ui-react";

import { createFragmentContainer } from "react-relay";

import { KeyListItem_item } from "./__generated__/KeyListItem_item.graphql";

import "./KeyListItem.css";

export interface IProps {
  item: KeyListItem_item;
  onEdit: (values: IProps) => any;
  onDelete: (values: IProps) => any;
}

export function KeyListItem(props: IProps) {
  const{ item: { name, value }, onEdit, onDelete } = props;
  const handleEdit = () => onEdit({ ...props });
  const handleDelete = () => onDelete({ ...props });

  return (
    <List.Item className="KeyListItem">
      <List.Content>
        <List.Header>{name}</List.Header>
        <List.Description>
          <code>
            {value}
          </code>
        </List.Description>
        <Button
          icon="edit"
          content="Edit"
          color="teal"
          size="small"
          onClick={handleEdit}
        />
        <Button
          icon="delete"
          color="pink"
          size="small"
          onClick={handleDelete}
        />
      </List.Content>
    </List.Item>
  );
}

export default createFragmentContainer(KeyListItem, graphql`
  fragment KeyListItem_item on Key {
    id
    name
    value
  }`,
);
