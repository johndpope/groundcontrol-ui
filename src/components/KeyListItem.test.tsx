// Copyright 2019 Stratumn
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

import { shallow } from "enzyme";
import React from "react";
import { mocked } from "ts-jest/utils";

import { mockQueryPropAttrs } from "../testing/relay";

import { KeyListItem } from "./KeyListItem";

const props = {
  item: {
    ...mockQueryPropAttrs(),
    id: "id",
    name: "name",
    value: "value",
  },
  onDelete: jest.fn(),
  onEdit: jest.fn(),
};

beforeEach(() => {
  mocked(props.onDelete).mockClear();
  mocked(props.onEdit).mockClear();
});

describe("<KeyListItem />", () => {

  it("renders correctly", () => {
    const wrapper = shallow(<KeyListItem {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("triggers onDelete when the delete button is clicked", () => {
    const wrapper = shallow(<KeyListItem {...props} />);
    wrapper.find("Button[icon='delete']").simulate("click");
    expect(props.onDelete).toBeCalledTimes(1);
    expect(props.onDelete).toBeCalledWith(props);
  });

  it("triggers onEdit when the edit button is clicked", () => {
    const wrapper = shallow(<KeyListItem {...props} />);
    wrapper.find("Button[icon='edit']").simulate("click");
    expect(props.onEdit).toBeCalledTimes(1);
    expect(props.onEdit).toBeCalledWith(props);
  });

});
