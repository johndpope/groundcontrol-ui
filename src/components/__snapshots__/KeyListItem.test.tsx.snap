// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<KeyListItem /> renders correctly 1`] = `
<ListItem
  className="KeyListItem"
>
  <ListContent>
    <ListHeader>
      name
    </ListHeader>
    <ListDescription>
      <code>
        value
      </code>
    </ListDescription>
    <Button
      as="button"
      color="teal"
      content="Edit"
      icon="edit"
      onClick={[Function]}
      size="small"
    />
    <Button
      as="button"
      color="pink"
      icon="delete"
      onClick={[Function]}
      size="small"
    />
  </ListContent>
</ListItem>
`;
